helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -142,21 +142,21 @@ void release(AssignableReplica replica) throws IllegalArgumentException {
 
     AssignableReplica removedReplica = partitionMap.remove(partitionName);
     removedReplica.getCapacity().entrySet().stream()
-        .forEach(entry -> updateCapacityAndUtilization(entry.getKey(), -1 * entry.getValue()));
+        .forEach(entry -> updateRemainingCapacity(entry.getKey(), -1 * entry.getValue()));
   }
 
   /**
    * @return A set of all assigned replicas on the node.
    */
-  public Set<AssignableReplica> getAssignedReplicas() {
+  Set<AssignableReplica> getAssignedReplicas() {
```

Comment:
can you parallel stream this?

Reply:
Let's be aware of using parallelStream in non-expensive operations because parallelStream itself is more expensive than simple stream

Reply:
Since it is already inside a parallelStream in the caller's logic, do parallel here will slow it down dramatically.